code/modules/mob/living/simple_animal/bot/honkbot.dm

```diff
@@ -226,33 +224,32 @@
 				playsound(loc, 'sound/misc/sadtrombone.ogg', 25, 1, -1)
 				back_to_idle()
 				return
-			if(target)		// make sure target exists
-				if(Adjacent(target) && isturf(target.loc))
-					if(threatlevel <= 4)
-						honk_attack(target)
-					else
-						if(threatlevel >= 6)
-							set waitfor = 0
-							stun_attack(target)
-							anchored = FALSE
-							target_lastloc = target.loc
-					return
-				else	// not next to perp
-					var/turf/olddist = get_dist(src, target)
-					walk_to(src, target, 1, 4)
-					if((get_dist(src, target)) >= (olddist))
-						frustration++
-					else
-						frustration = 0
-			else
+
+			if(!target)		// make sure target exists
 				back_to_idle()
+				return
+
+			if(Adjacent(target) && isturf(target.loc))
+				if(threatlevel <= 4)
+					honk_attack(target)
+				else
+					if(threatlevel >= 6)
+						set waitfor = 0
```

Comment:
not sure how I feel about this just being in a proc like this. It may be better just use INVOKE_ASYNC here instead and pull this stuff out into it's own proc